src/infra/core/src/utils/simple_transfer_protocol.rs

```diff
@@ -214,6 +240,143 @@ impl SimpleTransferProtocol {
         }
     }
 
+    async fn downlaod_block_data<'a>(
+        &'a self,
+        src: &'a dyn Dataset,
+        dst: &'a dyn Dataset,
+        data_slice: &DataSlice,
+        trust_source_hashes: bool,
+        listener: Arc<dyn SyncListener>,
+        mut stats: SyncStats,
+    ) -> Result<(), SyncError> {
+        tracing::info!(hash = ?data_slice.physical_hash, "Transfering data file");
+
+        let stream = match src
+            .as_data_repo()
+            .get_stream(&data_slice.physical_hash)
+            .await
+        {
+            Ok(s) => Ok(s),
+            Err(GetError::NotFound(e)) => Err(CorruptedSourceError {
+                message: "Source data file is missing".to_owned(),
+                source: Some(e.into()),
+            }
+            .into()),
+            Err(GetError::Access(e)) => Err(SyncError::Access(e)),
+            Err(GetError::Internal(e)) => Err(SyncError::Internal(e)),
```

Comment:
Proposition: extract error mapping logic into `impl From<GetError> for SyncError`

For example, like `impl From<ContainsError> for GetError`